### DISCUSSION TITLE
Add collaborator

### DISCUSSION THREAD

**Discussion by Jayantadeb73** [community/community#21748] [New to GitHub]
<p>Hi All,</p>
<p>I am new in GitHub and just started palying around to gain knowledge on this tool.</p>
<p>I have done followings :</p>
<ol>
<li>
<p>Created one private github account </p>
</li>
<li>
<p>Created one public repository</p>
</li>
</ol>
<p>Now i would like add collaborator to this new repo.</p>
<p>For this i have done followings:</p>
<ol>
<li>
<p>Select ‘Settings’ options of the repository for which i would like to invite collabirator</p>
</li>
<li>
<p>Click on ‘Collaborator’ option</p>
</li>
<li>
<p>in ‘Add Collaborator’ text box, find and added on of my freind’s git hub account</p>
</li>
<li>
<p>Click on ‘Add Collaborator’ button</p>
</li>
<li>
<p>Now it is showing awating for response from the user whome i have sent the invitation.</p>
</li>
</ol>
<p>now, problem is neither my friend is geeting any mail from my end regarding this invitation nor he is geeting any notification in his github account, hence unable to add collaborator to my repo.</p>
<p>Don’t know what to do in this case.</p>
<p>Looking for your guidance.</p>
<p>Thanks with Regards,</p>
<p>JD </p>

**Comment by mpboom** [comment]
<p>Hi <a class="mention" href="/jayantadeb73">@jayantadeb73</a> ,</p>
<p>Did your friend, who you invited, try to:</p>
<ol>
<li>Log in with their account</li>
<li>Manually navigate to your private repo</li>
</ol>
<p>If the invite got sent out well, a screen to accept the invite should appear.</p>
<p>Also check the junk folder in the mailbox! Post back if it didn’t work.</p>

**Comment by Jayantadeb73** [comment]
<p>Hi Mark,</p>
<p>Thanks for your reply.</p>
<p>As mentioned two steps in below mail, i can follow the first one, but unable to follow 2nd one (as you know i am very new in GitHub).</p>
<p>It will be nice if you can let me know the how to ‘manually navitate’ to my private repo from my friend’s account.</p>
<p>Sorry to bother you.</p>
<p>Thanks with Regards,</p>
<p>JD</p>

**Comment by mpboom** [comment]
<p>No problem <a class="mention" href="/jayantadeb73">@jayantadeb73</a> , I should have explained it better!</p>
<p>Normally, the URL of a repository would be <strong><a href="http://github.com/" rel="nofollow noopener">github.com/</a>&lt;username&gt;/&lt;repo name&gt;</strong>. If I were to navigate to a private repository by typing it manually in the address bar, I’d get a 404. However, if I am invited to that private repository I <strong>should</strong>  get a screen where I can accept the invite.</p>
<p>So, if your repo is called <strong>test</strong> , first invite your friend as a collaborator, then let him sign into that account and let him navigate to <strong><a href="http://github.com/Jayantadeb73/test" class="inline-onebox-loading" rel="nofollow noopener">github.com/Jayantadeb73/test</a></strong>  (make sure to replace the repo name!).</p>
<p>Let us know how it goes!</p>

**Comment by Jayantadeb73** [comment]
<p>Hi Mark,</p>
<p>Thank you very much for such a excellent explanation regarding ‘navigate to private repo’; but unfortunately didn’t work as expected.</p>
<p>Now, what i did, i have created another reposiroty, this time as private - (earlier one was public for which i have already sent invitation, which didn’t work, even after followed your instruction) -  and sent invitation for this private repository.</p>
<p>Now one again followed your instruction and this time worked  as per expectation… my friend can see the invitation which he has accepted.</p>
<p>Now here i have couple of queries:</p>
<ol>
<li>
<p>why the invitation didn’t work for public repo, but work for private. </p>
</li>
<li>
<p>even for ‘private repo’ it wokrs after followed your insrutction, but why he has not received any notification in his github account. but received the invitation mail.</p>
</li>
<li>
<p>and finally I am not able to provide the access rights (admin, read, write), not getting option even fot this… </p>
</li>
</ol>
<p>Looking for you kind reply.</p>
<p>Thanks with Regards,</p>
<p>JD</p>

**Comment by mpboom** [comment, marked as answer]
<p>As a helpful reference, here is the official article on it: <a href="https://help.github.com/en/articles/inviting-collaborators-to-a-personal-repository" rel="nofollow noopener">https://help.github.com/en/articles/inviting-collaborators-to-a-personal-repository</a></p>
<p>To (try to) answer your questions:</p>
<ol>
<li>Not sure about this, but I don’t think adding people as collaborators on public repos will have much of an effect in like mails and such.</li>
<li>I don’t think you every receive notifications inside of your account if you get invited to any repo.</li>
<li>If you’ve added your friend as a collaborator, he should have push rights. Just ask him to try out to push a branch.</li>
</ol>
<p>Initially in this thread, I thought you were talking about private repos as well. In my experience, it works a bit along these lines:</p>
<ul>
<li>Inviting someone to a public repo just “silently” (not sure about that) gives that person the rights to push</li>
<li>Inviting someone to a private repo sends out mails and such</li>
</ul>
<p>Some things for you to check:</p>
<ul>
<li>Is your friend able to push to your public repo after you’ve added them as a collaborator (in which case the sytem works)</li>
<li>If you are trying to add collaborators to a private repo, remember that using Free accounts you can only have up to three collaborators per  <strong>private</strong> repository</li>
<li>If this fails as well, I’d suggest contacting GitHub support at <a href="https://github.com/contact" rel="nofollow noopener">https://github.com/contact</a>, since I can’t look into your account</li>
</ul>

**Comment by Jayantadeb73** [comment]
<p>Hi Mark,</p>
<p>Thank you very much for your great support. This was my first conversation in GitHub account and you made it special. Once again thanks for all your answers.</p>
<p>Definitely i will look into all those matters, mentioned by you and let you know if any hurdles come.</p>
<p>Thanks with Rergards </p>
<p>JD</p>
